### PR TITLE
Add Guild Member avatar

### PR DESCRIPTION
**Pull request by andresperezl**
Fixes https://github.com/bwmarrin/discordgo/issues/965

This PR adds support for the Guild-specific member avatar, if any.

To follow the same pattern as `*User.AvatarURL`, also add the same method for `*Member.AvatarURL` , and create an util method `avatarURL` to handle both.

Also, set the `*Member.GuildID` after the `GuildMember` requests, as this is not included with the returned object.

I have tested these changes with my own bot

### CODE REVIEW

**Comment by andresperezl** [comment]
I have tested these changes with my own bot

**Comment by andresperezl** [comment]
I see this is a duplicate of #985, but the original author seems to be inactive after being asked for some changes

w.r.t https://github.com/bwmarrin/discordgo/pull/985#discussion_r769135345

I saw other `string` fields which are also nullable to be left as simple `string` (like `Nick`) so I followed the same, but I can change it to `*string` if that is what is desired.

cc @FedorLap2006 @CarsonHoffman

File: restapi.go [hunk 1]
```diff
@@ -790,6 +790,8 @@ func (s *Session) GuildMember(guildID, userID string) (st *Member, err error) {
 	}
 
 	err = unmarshal(body, &st)
+	// The returned object doesn't have the GuildID attribute so we will set it here.
+	st.GuildID = guildID
```
Comment (FedorLap2006) [inline restapi.go:794]: Good catch! Not sure tho if it would be the right PR to include the fix.

File: structs.go [hunk 2]
```diff
@@ -794,6 +794,9 @@ type Member struct {
 	// Whether the member is muted at a guild level.
 	Mute bool `json:"mute"`
 
+	// The avatar of the member for the guild, if any.
+	Avatar string `json:"avatar"`
```
Comment (FedorLap2006) [inline structs.go:798]: The doc-comment might be a little confusing. Since this is a hash, not the avatar url.

**Review by FedorLap2006 — CHANGES_REQUESTED** [review]
Overall looks good to me. But it requires a couple of changes.

File: restapi.go (hunk 1 above)
Comment (andresperezl) [reply restapi.go:794]: This is actually needed for the `*Member.AvatarURL` method, else it wouldn't be able to form the URL https://github.com/bwmarrin/discordgo/pull/1077/files#diff-ed32211335d38b2d817f745b55cd71fcc6f824cc578a680cdbed3912002876b6R829, but I can move it if necessary

Comment (FedorLap2006) [reply restapi.go:794]: Oh, then it's all good. No need to move it to different PR.

**Review by FedorLap2006 — APPROVED** [review]

**Comment by FedorLap2006** [comment]
Tip: you can resolve conversations when they're outdated or finished, anyways, thank you for the PR. +1

**Comment by FedorLap2006** [comment]
It's ready to be merged, but please note the conflict.

**Comment by andresperezl** [comment]
@FedorLap2006 I have rebased my changes with latest `master`

File: structs.go (hunk 2 above)
Comment (FedorLap2006) [reply structs.go:929]: Still no change in the comment. Maybe you forgot to push?

File: structs.go [hunk 3]
```diff
@@ -925,6 +925,9 @@ type Member struct {
 	// Whether the member is muted at a guild level.
 	Mute bool `json:"mute"`
 
+	// The avatar of the member for the guild, if any.
```
Comment (FedorLap2006) [inline structs.go:928]: The comment is still unchanged.

**Review by FedorLap2006 — CHANGES_REQUESTED** [review]

**Review by FedorLap2006 — APPROVED** [review]

**Comment by FedorLap2006** [comment]
Thanks for the contribution!

**Pull request merged**